pkg/common/proto/agent.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+service AgentService{
+    rpc PerformFantasyActions(FantasyLandscape) returns (FantasySelections) {}
+}
+
+message FantasyLandscape{
+    int32 matchNumber = 1;
+    LeagueSettings settings = 2;
+    FantasyTeam botTeam = 3;
+    repeated Player players = 4;
+    /* TODO historical / statistical data */
+    /* TODO competition lineups */
+}
+
+message FantasySelections{
+    repeated PlayerSlot slots = 1;
+}
+
+message FantasyTeam{
+    string id = 1;
+    string name = 2;
+    string owner = 3;
+}
+
+message LeagueSettings{
+    int32 numTeams = 1;
+    repeated PlayerSlot slotsPerTeam = 2;
+    /* scoring, budgets, etc. */
+}
+
+message PlayerSlot{
+    string name = 1;
+    bool canBeEmpty = 2;
```

Comment:
Out of curiousity, what is the use case for this field? I'm imagining that basically any slot could be empty.

Reply:
ya this was mainly thinking about a use case where it's invalid to not have a play rostered...might be too early though. I'll remove it for now